Src/Sarif.PatternMatcher/SearchSkimmer.cs

```diff
@@ -33,7 +31,8 @@ public class SearchSkimmer : Skimmer<AnalyzeContext>
         public readonly IList<MatchExpression> MatchExpressions;
 
         private const string DefaultHelpUri = "https://github.com/microsoft/sarif-pattern-matcher";
-        private const string Base64DecodingFormatString = "\\b(?i)[0-9a-z\\/+]{0}";
+        private const string Base64DecodingFormatString = "\\b(?i)[0-9a-z\\/+]{{{0},{1}}}[=]{{0,2}}";
+        private const string Base64DecodingRegex = "^(?i)(?:[0-9a-z\\/+]{4})*(?:[0-9a-z\\/+]{2}==|[0-9a-z\\/+]{3}=)?$";
```

Comment:
>^ [](http://example.com/codeflow?start=52&length=1)

Do we need to assert start of line and end of line for this regex?

Reply:
here its used to match the whole string if a valid base64.
without the start/end line, it will find some false-positives.
e.g. "abcdef" is an invalid base64 missing padding "==" at the end.
without the start/end line check, it will find a match on "abcd".